vulnerable_people_form/integrations/persistence.py

```diff
@@ -0,0 +1,199 @@
+    )
+
+
+def _find_database_arn(app):
+    rds = get_rds_client(app=app)
+    clusters = rds.describe_db_clusters()["DBClusters"]
+    coronavirus_rds = [
+        x
+        for x in clusters
+        if x["DBClusterIdentifier"].startswith(app.config["DATABASE_CLUSTER_PREFIX"])
+    ][0]
+    return coronavirus_rds["DBClusterArn"]
+
+
+def _find_database_secret_arn(app):
+    secrets_manager = get_secretsmanager_client(app=app)
+    return secrets_manager.list_secrets(
+        Filters=[{"Key": "tag-value", "Values": app.config["DATABASE_SECRET_TAGS"]}]
+    )["SecretList"][0]["ARN"]
+
+
+def init_app(app):
+    if app.config.get("AWS_RDS_DATABASE_ARN_OVERRIDE"):
+        app.config["AWS_RDS_DATABASE_ARN"] = app.config["AWS_RDS_DATABASE_ARN_OVERRIDE"]
+    else:
+        app.config["AWS_RDS_DATABASE_ARN"] = _find_database_arn(app)
+
+    if app.config.get("AWS_RDS_SECRET_ARN_OVERRIDE"):
+        app.config["AWS_RDS_SECRET_ARN"] = app.config["AWS_RDS_SECRET_ARN_OVERRIDE"]
+    else:
+        app.config["AWS_RDS_SECRET_ARN"] = _find_database_secret_arn(app)
+
+
+def generate_string_parameter(name, value):
+    return {
+        "name": name,
+        "value": {"isNull": True} if value is None else {"stringValue": value,},
+    }
+
+
+def generate_int_parameter(name, value):
+    return {
+        "name": name,
+        "value": {"isNull": True} if value is None else {"doubleValue": value,},
+    }
+
+
+def generate_bigint_parameter(name, value):
+    return {
+        "name": name,
+        "value": {"isNull": True} if value is None else {"longValue": value,},
+    }
+
+
+def generate_date_parameter(name, value):
+    return {
+        "name": name,
+        "typeHint": "DATE",
+        "value": {"isNull": True}
+        if value is None
+        else {"stringValue": "{year}-{month:0>2}-{day:0>2}".format(**value),},
+    }
+
+
+def execute_sql(sql, parameters):
+    return get_rds_data_client().execute_statement(
+        sql=sql,
+        parameters=parameters,
+        resourceArn=current_app.config["AWS_RDS_DATABASE_ARN"],
+        secretArn=current_app.config["AWS_RDS_SECRET_ARN"]
+    )
+
+
+def persist_answers(
+    nhs_number,
+    first_name,
+    middle_name,
+    last_name,
+    date_of_birth,
+    address_line1,
+    address_line2,
+    address_town_city,
+    address_county,
+    address_postcode,
+    address_uprn,
+    contact_number_calls,
+    contact_number_texts,
+    contact_email,
+    uid_nhs_login,
+    are_you_applying_on_behalf_of_someone_else,
+    have_you_received_an_nhs_letter,
+    do_you_want_supermarket_deliveries,
+    do_you_need_help_meeting_your_basic_care_needs,
+    do_you_have_any_special_dietary_requirements,
+    do_you_have_someone_in_the_house_to_carry_deliveries,
+    do_you_have_one_of_the_listed_medical_conditions,
+):
+    result = execute_sql(
+        sql="CALL cv_staging.create_web_submission("
+        ":nhs_number,"
```

Comment:
Handle the potential errors on `execute_sql` looks important here

I expect you should probably think about handling all the boto3 errors but here seems particularly important to avoid data loss

Reply:
What happens if the `CALL` returns a message/error state indicating that the query failed validation or something? Will that also be handled?

Reply:
That will bubble up to Flask, and the user's POST request will be redirected to an error page. The per-page form validation should make it impossible to have invalid data when submitting the 'whole' form, so I think this behaviour is legitimate. In addition, the transaction will exit with a rollback.

Reply:
@danwilliams we don't currently have a Sentry instance provisioned since we can't use the instance used by GOV.UK which the Rails app was using. I am trying to get a new instance approved and set up at the moment, but we can't currently rely on Sentry being in place.

Reply:
sentry has been removed from the app in the concourse pipeline branch as @pipwilson has pointed out that there is no instance available to use

Reply:
Correct, as per my earlier answers this behaviour is not dependent upon Sentry - it seems we have removed that elsewhere, which is fine, but out of scope of these changes 🙂

Reply:
If we go with Flask as-is what will be catching doing our logging?
Just a guess but I'd think with the PaaS out of the box the answer is no
I think that question needs answering but outside of this PR